gradle/check.gradle

```diff
@@ -69,13 +69,14 @@ quickCheck.dependsOn checkstyleMandatory
 apply plugin: 'com.github.spotbugs'
 
 spotbugs {
-    toolVersion = '3.1.7'
+//  toolVersion = '4.2.1'
     ignoreFailures = false
     effort = 'max'
-    sourceSets = [sourceSets.main]
     excludeFilter = rootProject.file('gradle/codequality/spotbugs_exclude.xml')
 }
 
+spotbugsTest.enabled = false
```

Comment:
I think there may be a *fourth* way of excluding files from SpotBugs that may allow us to turn `spotbugsTest` back on, which is by using the `Source` filter in exclude file. At the moment, that's set to exclude all classes with `Proto` in the name:

https://github.com/FoundationDB/fdb-record-layer/blob/8382bf1a5bc172594e1d18a2630c9a603abf99ca/gradle/codequality/spotbugs_exclude.xml#L5-L7

(Which may be why `spotbugsMain` isn't hitting the same problems as `spotbugsTest`? That is, the filter may be correctly excluding all non-test generated protos but missing (some of?) the test generated protos?) In theory, I think we could use the [`<Source>`](https://spotbugs.readthedocs.io/en/stable/filter.html#source) tag to filter out the `protogen` files.

I also think merging this with `spotbugsTest` off and then re-enabling it if/when we figure out the right way to exclude those generated files would make sense.

Reply:
Or maybe that regex should be `"~.*Proto\$?.*"` to match the filtering that's been removed?

Reply:
Using `Source` is clearer on intent, yes. So I switched to that.

There are actual failures in `spotBugsTest`. I suppose fixing them and keeping it enabled might be worthwhile at some point.

Reply:
I see. It would certainly be good to get that re-enabled, but that sounds like a separate thing from this PR. I filed #1204 to track that.